File: level76/homework/codwars8.py
#https://www.codewars.com/kata/597a660f59873cc353000061/train/python (RGB 3)
def resolve_hex(hex):
    r, g, b = hex[0:2], hex[2:4], hex[4:]
    if r < g > b:
        return "Green"
    elif g < r > b:
        return "Red"
    else:
        return "Blue"


def rank_colors(colors):
    return sorted(list(set(colors)), key=colors.count, reverse=True)


def get_colors(col_arr):
    ranked_colors = ('+'.join(rank_colors([resolve_hex(hex)
                                           for hex in hex_list]))
                     for hex_list in col_arr)
    return ','.join(ranked_colors)

File: level 076/homework/codwars8.js
//https://www.codewars.com/kata/597a660f59873cc353000061/train/javascript
function getColors(colArr) {
    let result = []
    for (let i = 0; i < colArr.length; i++) {
        let rgb = {}
        for (let j = 0; j < colArr[i].length; j++) {
            let tmp = {}, max
            let color = colArr[i][j]
            tmp.Red = parseInt(color.slice(0,2),16)
            tmp.Green = parseInt(color.slice(2,4),16)
            tmp.Blue = parseInt(color.slice(4,6),16)
            max = Object.keys(tmp).reduce((p, c) => tmp[p] > tmp[c] ? p : c)
            rgb[max] = rgb[max] + 1 || 1
        }
        let maxMin = Object.keys(rgb).reduce((p, c) => rgb[p] > rgb[c] ? p : c) + '+' + Object.keys(rgb).reduce((p, c) => rgb[p] < rgb[c] ? p : c)
        result.push(maxMin)
    }
    return result.join()
}